File: Nololiyt.IndexedDbSharp.Raw/Nololiyt.IndexedDbSharp.Raw/Typescripts/Wrapping/WrappedDomException.ts
export class WrappedDomException
{
    private wrapped;
    constructor(wrapped: DOMException)
    {
        this.wrapped = wrapped;
    }

    message(): string
    {
        return this.wrapped.message;
    }

    name(): string
    {
        return this.wrapped.name;
    }

    stack(): string | undefined
    {
        return this.wrapped.stack;
    }
    
    code(): number
    {
        return this.wrapped.code;
    }
}

File: Nololiyt.IndexedDbSharp.Raw/Nololiyt.IndexedDbSharp.Raw/CSharp/Async/Implementation/Wrapping/WrappedDomException.cs
using Microsoft.JSInterop;
using Nololiyt.IndexedDbSharp.Raw.CSharp.Async.Wrapping;

namespace Nololiyt.IndexedDbSharp.Raw.CSharp.Async.Implementation.Wrapping
{
    internal sealed class WrappedDomException :  WrappedWrappedJsObjectBase, IWrappedDomException
    {
        public WrappedDomException(IJSObjectReference wrappedObject) : base(wrappedObject)
        {
        }

        public async ValueTask<int> GetCodeAsync()
        {
            return await this.WrappedObject.InvokeAsync<int>("code");
        }

        public async ValueTask<string> GetMessageAsync()
        {
            return await this.WrappedObject.InvokeAsync<string>("message");
        }

        public async ValueTask<string> GetNameAsync()
        {
            return await this.WrappedObject.InvokeAsync<string>("name");
        }

        public async ValueTask<string?> GetStackAsync()
        {
            return await this.WrappedObject.InvokeAsync<string?>("stack");
        }
    }
}
